doc/development/guide/tests.rst

```diff
@@ -16,26 +15,126 @@ These requirements can be installed via ``pip``:
 
     pip install pytest pytest-cov pytest-mock flaky
 
+Creating a test
+~~~~~~~~~~~~~~~
+Every test has to be added to the PennyLane `test folder <https://github.com/PennyLaneAI/pennylane/tree/master/tests>`__.
+The test folder is following PennyLane structure, and therefore a test needs to be added to its corresponding folder.
+
+Most tests do not require any interface (Autograd, Torch, TensorFlow and Jax). Tests without interface will be marked
+as ``core`` tests automatically by pytest ``conftest.py``. For general tests, you can follow this example structure,
+where it is only required to follow pytest guidelines:
+
+.. code-block:: python
+
+    import pennylane as qml
+
+    def test_circuit_expval(self):
+        """ Test that the circuit expectation value for PauliX is 0."""
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.qnode(dev)
+        def circuit():
+            return qml.expval(qml.PauliX(0))
+
+        expval = circuit()
+
+        assert expval == 0
+
+This test will be marked automatically as a ``core`` test.
+
+On the other hand, some tests require specific interfaces and need to marked in order to be run on our Github test suite.
+Tests involving interfaces have to be marked with its respective marker: ``@pytest.mark.autograd``, ``@pytest.mark.torch``,
+``@pytest.mark.tf`` and ``@pytest.mark.jax``. It is now prohibited to use  `ìmportorskip()`` inside tests with interfaces.
```

Comment:
Minor question: what happens if you have a test that uses more than one interface?

Reply:
I used the marker ``pytest.mark.math`` for mixed interfaces, because the math module consists of mixed interfaces. It has a dedicated runner on GIthub where all interfaces are installed. I will detail that here.

Reply:
Nice, and tests marked as such will be run by _all_ CI actions? I wonder if `pytest.mark.all` might help generalize it beyond the `math` module?

Reply:
No it has a dedicated runner called ``math`` and is only run once. But I agree that ``all`` could be a great marker.